code-dot-org/code-dot-org
# Assign course to section(s) from curriculum catalog

When a signed-in teacher has sections, they can now assign courses and units from our catalog page!

This card is specifically for signed-in teachers with sections. See follow-up work for more.

I did need to apply some inline styles to the MultipleSectionsAssigner component to make the right/left styling compatible with RTL pages. No other changes were necessary!

What the component looks like from the unit overview page in production:
<img width="505" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/42243189-5833-4d3e-9c76-4c3a76337d61">

**See screenshot in comment at the bottom for the newest version of these –– placement of the horizontal line changed, as well as padding on the Select All**
What the component looks like from the unit overview page (should be unchanged from what's above):
<img width="479" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/7aa86602-f659-4991-bdcd-50b3fdbb71d5">

What the component looks like in RTL languages:
<img width="477" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/037a2159-3e5a-46b5-9e8d-c2f128e935d5">

From the catalog page, there is a pesky underline in the <a> tag –– I've created a follow-up ticket for this issue, as this tag should be a `button` for accessibility (it's not a link to a new page)
<img width="514" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/593a98d0-8a44-4eef-83ce-cd11e1af43d5">


## Links





## Testing story

I added a unit test for this, but I think the bulk of the test coverage will come from UI and/or eyes tests

### Manual testing:

**Screenshots are from before I disentangled the catalog styling from the dialog styling! See updated styling above**

My local teacher account started with the following sections assigned:
<img width="805" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/60add027-b1f4-421f-816b-1216b66b9772">

When I go to the curriculum catalog card and click on "Course 1", I see:
<img width="803" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/67f27c15-7ac9-4691-9b59-bce51411203b">

If I uncheck "test", the course is un-assigned:
<img width="803" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/35a6e44c-0ec7-45f1-a389-7be73dca3065">

When I go to the curriculum catalog card and click on "Course 2", I see:
<img width="801" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/71db0d9b-ba9b-44f9-9e33-8856b3f0c756">

If I check both sections, the course is assigned to both:
<img width="829" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/c47badba-63de-4adc-a598-81027410fab9">

And if I select a totally different course and assign it:
<img width="801" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/ab88878c-45d3-4f0f-8245-aa2ae3bb281c">

I see:
<img width="814" alt="image" src="https://github.com/code-dot-org/code-dot-org/assets/9142121/2c77efac-7c45-423a-9a13-222d0a691e77">


## Deployment strategy

## Follow-up work

- We're still getting the final spec for a confirmation message for when a course was successfully assigned –– will follow up with that in a separate PR. Ticket here https://codedotorg.atlassian.net/browse/ACQ-643
- Need to add keyboard accessibility to the dialog component (https://codedotorg.atlassian.net/browse/ACQ-553)
- Need to add functionality for signed-out users, and for signed-in teachers who do not have sections (https://codedotorg.atlassian.net/browse/ACQ-521 and https://codedotorg.atlassian.net/browse/ACQ-621)
- The "Select All" is currently an anchor tag, presumably for styling purposes. Unfortunately, that's not accessible, as the functionality is a button. Also unfortunately, with CSS-in-JS styles, you can't add styles specific to state (e.g. hover) :(. So it's not a quick fix. Ticket here: https://codedotorg.atlassian.net/browse/ACQ-642

## Privacy



## Security



## Caching

## PR Checklist:
